netflix_3.0_word2vec.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import sys
+import numpy
+from gensim.models.keyedvectors import KeyedVectors
+from math import log10
+
+from netflix_utils import NetflixUtils
+import netflix_data
+
+import tensorflow as tf
+
+BATCH_SIZE = 512
+DEFAULT_N_ITER = int(40 * netflix_data.NUM_RATING / BATCH_SIZE)
+REGULARIZATION_FACTOR = 0.5
+LEARNING_SPEED = 0.5
+MODEL_NAME = "netflix_0.0_latent"
+
+def preprocess_word_nt(movie_titles):
+    nt = {}
+    for movie_title in movie_titles:
+        for word in movie_title:
+            if word in nt:
+                nt[word] = nt[word] + 1
+            else:
+                nt[word] = 1
+    return nt
+
+def preprocess_title_embeddings(model, movie_titles, nt, ncol):
+    embeddings = numpy.zeros((len(movie_titles), ncol))
+    for i, movie_title in enumerate(movie_titles):
+        embedding = numpy.zeros((ncol))
+        length = 0
+        for word in movie_title:
+            if word in model.wv.vocab:
+                word_vector = model[word][0:ncol]
+                tfidf = log10(len(movie_titles) / nt[word])
+                embedding = embedding + tfidf * word_vector
+                length = length + 1
+        embedding = embedding / length
+        embeddings[i, :] = embedding
+
+    return embeddings
+
+def preprocess_get_titles(utils):
+    if utils.args.gen_movie_titles:
+        all_titles = []
+        with open('nf_prize_dataset/movie_titles.tsv') as f:
+            first = True
+            for line in f:
+                if first: #ignore header line
+                    first = False
+                    continue
+                words = line.split()
```

Comment:
Nit: line.split()[2] instead of 2 pop I guess

Reply:
Pas exactement, vu que ya 3 *ou plus* mots. Mais j'avoue que je n'ai plus besoin de pop deux fois maintenant que je n'ai pu besoin du ID, je vais changer ça

Reply:
Ack

Reply:
i est jamais incrémenté?

Reply:
Nvm